### PR TITLE
88 supporting document uploads

### PR DESCRIPTION
**Pull request by marcellmueller**
I wouldn't normally do this but I ended up combing three tickets in this one since they are all directly related in that they required the custom `FileWidget` I created for #88. So this PR contains the new file widget and three new pages, `template uploads`, `supporting documents`, and `mapping`.

Another thing to note is that since we don't have file uploads working the requirements for the file widget was just the html. Though I did make the file picker work and it currently displays the file and saves the file name in the form.

https://app.zenhub.com/workspaces/ccbc---delivery-board-61c0c1204c8bcb001491c5f3/issues/bcgov/conn-ccbc-portal/88
https://app.zenhub.com/workspaces/ccbc---delivery-board-61c0c1204c8bcb001491c5f3/issues/bcgov/conn-ccbc-portal/87
https://app.zenhub.com/workspaces/ccbc---delivery-board-61c0c1204c8bcb001491c5f3/issues/bcgov/conn-ccbc-portal/89

### CODE REVIEW

**Comment by AntBush** [comment]
Other than the file type restrictions, everything looks good to me

File: app/lib/theme/widgets/FileWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      <Button
+        id={id}
+        onClick={(e: React.MouseEvent<HTMLInputElement>) => {
+          e.preventDefault();
+          handleClick();
+        }}
+      >
+        Upload
+      </Button>
+      <input
```
[the remark this answers is not in the document]
Comment (marcellmueller) [reply app/lib/theme/widgets/FileWidget.tsx:68]: @AntBush likely though this is just the html for this card, I just wanted to make sure the rjsf field was saving with this custom input.

[the remark this answers is not in the document]
Comment (AntBush) [reply app/lib/theme/widgets/FileWidget.tsx:68]: Ah yeah, want to leave a `//TODO` to keep that in mind?

[the remark this answers is not in the document]
Comment (marcellmueller) [reply app/lib/theme/widgets/FileWidget.tsx:68]: If you feel strongly about it but realistically the whole file is a Todo and very much an empty shell that requires file uploads, file size limitations, file type limitations, extra functionality to delete a file that has been uploaded, the ability to upload multiple files (this one might be hard in rjsf).

**Comment by marcellmueller** [comment]
> Other than the file type restrictions, everything looks good to me

File type restrictions will need a lot more consideration and doesn't belong in this ticket which just wanted the basic design in the form. We will have to make a custom `ui:file-type` prop or similar to specify file types per field and know more about the requirements.

**Pull request merged**